lib/util/metadata_tools.rb

```diff
@@ -0,0 +1,187 @@
+module MetadataTools
+
+  def add_dataset_metadata(obj, orgs_cache)
+    d = Dataset.find_or_create_by(uuid: obj["id"])
+    d.name = obj["name"]
+    d.title = obj["title"]
+    d.summary = generate_summary(obj["notes"])
+    d.description = obj["notes"]
+    d.organisation_id = orgs_cache[obj["owner_org"]]
+    d.licence = obj["license_id"]
+    d.licence = 'no-licence' if d.licence == ""
+    d.licence_other = ""
```

Comment:
There's a diff between no-licence and other :) We don't know what it is if it isn't specified so can't set to other.  This does need further work though for other licences we do know about.